### PR TITLE
build(dev_env): Exact version match for Python base requirements

### PR DESCRIPTION
**Pull request by armenzg**
This will reduce differences between each developers' set of dependencies as well
as differences with the CI.

This also makes `sentry` and `getsentry` dependencies closer to each other.

This is also a step toward allowing the getsentry bot to port commits from `sentry` to
`getsentry`, thus, reducing the need to coordinate PRs between the two repositories.

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -20,49 +20,50 @@ googleapis-common-protos==1.52.0
...
-psycopg2-binary>=2.7.0,<2.9.0
-PyJWT>=1.5.0,<1.6.0
+petname==2.6
+phonenumberslite==8.11.3
+Pillow==8.1.0
+progressbar2==3.32.0
+psycopg2-binary==2.7.7
+PyJWT==1.5.3
 pyrsistent==0.17.3
 python-dateutil>=2.0.0,<3.0.0
```
Comment (armenzg) [inline requirements-base.txt:35]: I will need to deal with this later as I need to bump the version on `getsentry` first.

File: requirements-base.txt [hunk 2]
```diff
@@ -20,49 +20,50 @@ googleapis-common-protos==1.52.0
...
 requests-oauthlib==1.2.0
-requests[security]>=2.20.0,<2.21.0
+requests[security]==2.20.1
 # [start] jsonschema format validators
 rfc3339-validator==0.1.2
 rfc3986-validator==0.1.1
 # [end] jsonschema format validators
-sentry-relay>=0.8.3,<0.9.0
+sentry-relay==0.8.3
 sentry-sdk>=0.20.0,<0.21.0
```
Comment (armenzg) [inline requirements-base.txt:51]: I will deal with this on another occasion as I need to fix the `pip freeze` before & after logic on `getsentry`.

**Comment by joshuarli** [comment]
@armenzg I can take over this if you want, I've just started the Python 3.8 dependency audit and I'll have to reorganize getsentry dependencies a little bit, and in that process I can see this through as well.

**Comment by joshuarli** [comment]
Actually, I just verified this is accurate (with the exception of dateutil but I see your comment).

In getsentry:

```
$ cat lol.py
#!/usr/bin/env python
import fileinput

for line in fileinput.input():
    tokens = line.split()
    print(f"{tokens[1]}=={tokens[-1][:-1]}")

$ pipdeptree --warn silence -p sentry | grep '^  -' | ./lol.py
beautifulsoup4==4.7.1
billiard==3.6.3.0
boto3==1.10.49
botocore==1.13.50
celery==4.4.7
cffi==1.12.2
click==7.1.2
confluent-kafka==1.5.0
croniter==0.3.34
cssselect==1.0.3
cssutils==1.0.2
datadog==0.15.0
Django==1.11.29
django-crispy-forms==1.7.2
django-picklefield==1.0.0
django-sudo==3.1.0
djangorestframework==3.6.4
email-reply-parser==0.5.12
google-api-core==1.25.1
google-auth==1.24.0
google-cloud-bigtable==1.6.1
google-cloud-core==1.5.0
google-cloud-pubsub==2.2.0
google-cloud-storage==1.35.0
googleapis-common-protos==1.52.0
grpcio==1.35.0
hiredis==0.3.1
jsonschema==3.2.0
kombu==4.6.11
lxml==4.6.2
maxminddb==1.5.4
mistune==0.8.4
mmh3==2.3.1
mock==4.0.3
msgpack==1.0.0
parsimonious==0.8.0
petname==2.6
phabricator==0.7.0
phonenumberslite==8.11.3
Pillow==8.1.0
progressbar2==3.32.0
psycopg2-binary==2.7.7
PyJWT==1.5.3
pyrsistent==0.17.3
python-dateutil==2.4.2
python-memcached==1.59
python-u2flib-server==5.0.0
python3-saml==1.4.1
PyYAML==5.3
qrcode==6.1
rb==1.8
redis==3.3.11
redis-py-cluster==2.1.0
requests==2.20.1
requests-oauthlib==1.2.0
rfc3339-validator==0.1.2
rfc3986-validator==0.1.1
selenium==3.141.0
sentry-relay==0.8.3
sentry-sdk==0.20.3
setproctitle==1.1.10
simplejson==3.11.1
sqlparse==0.2.4
statsd==3.1
structlog==17.1.0
symbolic==8.0.3
toronado==0.1.0
ua-parser==0.10.0
unidiff==0.5.5
urllib3==1.24.2
uwsgi==2.0.18
zstandard==0.14.1
```

Then the same in sentry on this branch, and this is the diff:

image

**Review by joshuarli — APPROVED** [review]
https://github.com/getsentry/sentry/pull/24005#issuecomment-786299795

Approving, but I trust this to not be merged until we can see it succeed with a getsentry branch.

**Comment by armenzg** [comment]
> [#24005 (comment)](https://github.com/getsentry/sentry/pull/24005#issuecomment-786299795)
> 
> Approving, but I trust this to not be merged until we can see it succeed with a getsentry branch.

Thanks for the review!
I've merged master here one more time.

I thought we test against getsentry in some cases but I'm unsure if `getsentry dispatch` worked:
https://github.com/getsentry/sentry/actions/runs/597429218
image

In any case, I've run this command for [this getsentry PR](https://github.com/getsentry/getsentry/pull/5169):
`./bin/bump-sentry fbf059a276bbaac5708e5a7049e858344d0ad16c`

I see this in the output which matches the sha I intended:
`Obtaining sentry-dev from git+https://github.com/getsentry/sentry.git@fbf059a276bbaac5708e5a7049e858344d0ad16c#egg=sentry-dev (from -r requirements-sentry.txt (line 2))`

If that PR passes and this one I will merge it.

**Comment by armenzg** [comment]
I had looked in the wrong place, it seems that `getsentry` backend tests were scheduled:
image

**Pull request merged**